api/v1alpha1/externaldns_utils.go

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+// PublicProvider returns true if the dns provider is one of supported public providers
+func (e *ExternalDNS) PublicProvider() bool {
```

Comment:
Is "Public" significant here, or would it be reasonable to name this method something like `ValidProvider` or `SupportedProvider`?

Reply:
No, not significant, I just didn't come up with a better name. The idea of this method is to tell that "if you are in the OpenShift case you need to do something only if it's one of these providers". Because we don't leverage OpenShift features for `BlueCat` and `InfoBlox`.

`Supported`, I don't know, this may give an impression that `BlueCat` and `InfoBlox` are not supported. The same can be thought about `Valid`. WDYT? Do you have any other ideas?

Reply:
Maybe `AutoCredentialConfigProvider()` if the cloud provider is supported for automatic credential configuration.

Reply:
@arjunrn: sorry, didn't see your comment. I called it `ManagedCredentialsProvider()`. Also I moved it outside the api package to remove the unnecessary method from the "public" API.